`--ra-mux-server` argument applies to `ra-multiplex` and not `ra-multiplex-server`

Thank you for building this crate. It's very handy.
### Problem
```bash
$ ra-multiplex-server --ra-mux-server /path/to/custom/rust-analyzer &
$ ra-multiplex <...>
```
then I get:
```
[INFO ] [32830] client connected
[ERROR] [32830] client error: spawn ra instance

     Caused by:
         0: couldn't spawn rust-analyzer
         1: No such file or directory (os error 2)
```
Passing `--ra-mux-server` to `ra-multiplex-server` doesn't seem to work. `ra-multiplex` cannot find `rust-analyzer`.

### Solution

You have to pass it via the *client*. To fix:
```bash
$ ra-multiplex-server --ra-mux-server /path/to/custom/rust-analyzer &
$ ra-multiplex --ra-mux-server /path/to/custom/rust-analyzer <...>
```
### Thoughts

1. Does passing `--ra-mux-server` to the server do anything?  If not, should it error?
2. If `--ra-mux-server` *is* a valid option to the server: In the absence of a client `--ra-mux-server`, the server's configuration should take priority.  
3. Maybe we could pass `ra-mux-server` via the config?
4. Maybe this could be spelled out a bit more in the docs? We could kill two birds with one stone by adding a `--help`?

I am happy to submit PRs for any/all of these. I just wanted to ask first because the intended/correct behavior aren't clear to me.

--- a/src/bin/server/instance.rs
+++ b/src/bin/server/instance.rs
@@ -238,7 +238,14 @@ impl RaInstance {
             .stdout(Stdio::piped())
             .stderr(Stdio::piped())
             .spawn()
-            .context("couldn't spawn rust-analyzer")?;
+            .with_context(|| {
+                format!(
+                    "couldn't spawn rust-analyzer with command: `{}{}{}`",
+                    &key.server,
+                    if key.args.is_empty() { "" } else { " " },
+                    key.args.join(" ")
+                )
+            })?;
 
         let pid = child.id().context("child exited early, couldn't get PID")?;
         let stderr = child.stderr.take().unwrap();

--- a/src/bin/server/main.rs
+++ b/src/bin/server/main.rs
@@ -8,8 +8,9 @@
 
 use crate::client::Client;
 use crate::instance::InstanceRegistry;
-use anyhow::{Context, Result};
+use anyhow::{Context, bail, Result};
 use ra_multiplex::config::Config;
+use std::env;
 use tokio::net::TcpListener;
 use tokio::task;
 
@@ -22,6 +23,10 @@ mod lsp;
 async fn main() -> Result<()> {
     let config = Config::load_or_default().await;
     let registry = InstanceRegistry::new().await;
+
+    if env::args().nth(1).is_some() {
+        bail!("`ra-multiplex-server` does not accept any arguments. To configure it, edit config.toml or pass arguments to the `ra-multiplex` client");
+    }
 
     let listener = TcpListener::bind(config.listen).await.context("listen")?;
     loop {

